pkg/target/regolib/src.rego

```diff
@@ -192,4 +194,23 @@ matches_namespaces(match) {
 	has_field(match, "namespaces")
 	ns := {n | n = match.namespaces[_]}
 	count({input.review.namespace} - ns) == 0
-}
\ No newline at end of file
+}
+
+matches_nsselector(match) {
+	not has_field(match, "namespaceSelector")
+}
+
+matches_nsselector(match) {
+	has_field(match, "namespaceSelector")
+	ns := data["{{.DataRoot}}"].cluster["v1"]["Namespace"][input.review.namespace]
+	matches_namespace_selector(match, ns)
+}
+
+# Checks to see if a kubernetes NamespaceSelector matches a namespace with a given set of labels
+# A non-existent selector or labels should be represented by an empty object ("{}")
+matches_namespace_selector(match, ns) {
+	nslabels := ns.metadata.labels
```

Comment:
we need two `get_default()` invocations here, in case there is no metadata and/or labels field